src/catalog/src/durable/transaction.rs

```diff
@@ -834,6 +835,49 @@ impl<'a> Transaction<'a> {
         }
     }
 
+    /// Removes all system object mappings in `descriptions` from the transaction.
+    ///
+    /// Returns an error if any description in `descriptions` is not found.
+    ///
+    /// NOTE: On error, there still may be some items removed from the transaction. It is
+    /// up to the called to either abort the transaction or commit.
```

Comment:
This feels little odd since I don't think any other Transaction APIs do this, but I can see why for this API specifically we might want this behavior. I think most, if not all, callsites will use `?` on the error so the transaction will get aborted anyways, but just wanted to call this out.

Reply:
Yeah it's definitely a little odd, though pretty much every method that updates/removes/inserts multiple values in this file works this way. It's mostly because we know all callers use `?` and will abort out of the transaction on error, so we avoid unnecessarily restoring the previous state just to have the caller throw it out.

Maybe a better API would be something like `fn remove_system_object_mappings(mut self, descriptions: BTreeSet<SystemObjectDescription>) -> Result<Self, CatalogError>`. That way we consume and drop the transaction on an error.